ecpy_hqc_legacy/instruments/drivers/visa/cryomagnetics_cs4.py

```diff
@@ -25,16 +25,26 @@
 
 _ACTIVITY_DICT = {'To zero': 'SWEEP ZERO'}
 
-FIELD_CURRENT_RATIO = 0.20393 # in tesla/amp
 OUT_FLUC = 2e-4
 MAXITER = 20
 
 
 class CS4(VisaInstrument):
 
+    def __init__(self, connection_info, caching_allowed=True,
+                 caching_permissions={}, auto_open=True):
+        super(CS4, self).__init__(connection_info, caching_allowed,
+                                  caching_permissions)
+        self.field_current_ratio = connection_info['magnet_conversion']
```

Comment:
This will first raise a KeyError. Better to check the presence of the key in the dictionary